### ISSUE TITLE
Bluetooth SPP w/ Direwolf

### ISSUE THREAD

**Issue by na7q** [ge0rg/aprsdroid#160]
I'm trying to use APRSDroid with Direwolf on my RPi over Bluetooth, but my issue is the incomplete data sent. Not sure if it's normal, an issue with Direwolf or what. Here's what it looks like.
Setting that up with,
sudo rfcomm watch /dev/rfcomm0 22 socat -d -d tcp4:127.0.0.1:8001 /dev/rfcomm0

Receiving packets is perfect on the phone. Sending looks as shown below in the pic in pink.
![img_0032](https://cloud.githubusercontent.com/assets/1294292/24130780/cd05feea-0da7-11e7-8f64-5e7aad884a93.PNG)

**Comment by ge0rg** [comment]
What are the APRSdroid connection settings you've configured? It should emit proper KISS AX.25 frames.

Can you also show a screenshot from the APRSdroid Log right after sending a packet?

**Comment by na7q** [comment]
I have the Bluetooth TNC settings all to default settings except with channel 22.
Here's the screenshots of what's happening.
![screenshot_2017-03-21-21-39-09](https://cloud.githubusercontent.com/assets/1294292/24182393/db913a42-0e7e-11e7-8851-6033c96d7d22.png)
![screenshot_2017-03-21-21-36-38](https://cloud.githubusercontent.com/assets/1294292/24182397/e6463ec4-0e7e-11e7-8125-690597273e2a.png)

**Comment by na7q** [comment]
Verbose output from socat. For sent and received packet.
![screenshot_2017-03-21-22-13-53](https://cloud.githubusercontent.com/assets/1294292/24183166/de92ae1a-0e83-11e7-90f3-7afbe7412396.png)

![screenshot_2017-03-21-22-14-17](https://cloud.githubusercontent.com/assets/1294292/24183159/ceebcbe0-0e83-11e7-8bbb-f07d9d2293b8.png)

**Comment by ge0rg** [comment]
Hi,

this is getting really interesting. APRSdroid is emitting the following via KISS protocol:

* `C0 00` KISS: FEND, Data frame
* 4x7 characters: callsigns: APDR13, NA7Q-7, WIDE1-1, WIDE2-1
* `03 F0` AX.25 UI-frame pid F0 no L3 protocol
* position and comment in ASCII
* `C0` KISS: FEND

If I didn't mistype your outgoing packet, it should generate the following bytes:

    c0 00
    82 a0 88 a4 62 66 e0 
    9c 82 6e a2 40 40 6e
    ae 92 88 8a 62 40 62
    ae 92 88 8a 64 40 63
    03 f0
    3d 34 36 31 30 2e 32 33 4e 2f 31 32 33 33 34 2e 32 36 57 3e 20 52 50 69 20 54 4e 43
    c0

Apparently direwolf dislikes something around the `03` byte, but I can't see what it is. Does any of the rfcomm interruptions happen during transmission? Do you have any non-ASCII characters in the APRS path setting on APRSdroid?

**Comment by na7q** [comment]
I'm not using any non-ASCII characters. I haven't noticed any interrruptions during transmission.
What is something I could get log wise that would better help you understand what's going on?
I can provide logcats of the app as well.

**Comment by na7q** [comment]
More info from Dire Wolf about the data frames
![img_0032](https://cloud.githubusercontent.com/assets/1294292/24226269/b5f2dd46-0f22-11e7-8a5f-8fb4ff530bba.PNG)

**Comment by na7q** [comment]
Here's a catlog screenshot of the app sending the data.
![screenshot_2017-03-22-17-18-51](https://cloud.githubusercontent.com/assets/1294292/24226475/c3533d72-0f23-11e7-951f-29748a0063e8.png)
![screenshot_2017-03-22-17-16-44](https://cloud.githubusercontent.com/assets/1294292/24226481/ca33b3ec-0f23-11e7-8b35-d3957bb83fa5.png)

**Comment by na7q** [comment]
Here are my final results. After getting some time to look everything over and with your help on the data frames. I tinkered with rfcomm and I found a solution that worked. I switched rfcomm to rfcomm --raw and this delivered the packets in full without issue! Fully functional now.
sudo rfcomm --raw watch /dev/rfcomm0 22 socat -d -d -v tcp4:127.0.0.1:8001 /dev/rfcomm0
![img_0040](https://cloud.githubusercontent.com/assets/1294292/24226874/6b95e686-0f26-11e7-914f-0a9097553f5a.PNG)
